### PR TITLE
[GUI] Replace thumb size with images res

### PR DESCRIPTION
**Pull request by Hitcher**
Replaces `GetThumbSize()` with `m_imageRes` so bookmark/chapter images
are the same resolution as posters, etc

Before -
Before
After-
After

@ronie @phil65 @popcornmix @Memphiz @MartijnKaijser @da-anda @Razzeee

### CODE REVIEW

**Comment by Memphiz** [comment]
in the end this doubles the thumbnail size right? Not into that discussion at all and have not much to say for this pr (technically its straight forward).

**Comment by HitcherUK** [comment]
It doubles the current method (which is half that used for posters, etc) for these thumbs only. All the other images are unaffected. This simply brings chapter/bookmark thumbs up to the same size as the rest.

File: xbmc/settings/AdvancedSettings.h [hunk 1]
```diff
@@ -237,7 +237,6 @@ class CAdvancedSettings : public ISettingCallback, public ISettingsHandler
     /*! \brief the maximal size to cache thumbs at, assuming square
      Used for actual thumbs (eg bookmark thumbs, picture thumbs) rather than cover art which uses m_imageRes instead
      */
-    unsigned int GetThumbSize() const { return m_imageRes / 2; };
```
Comment (da-anda) [inline xbmc/settings/AdvancedSettings.h]: you also have to remove the comment above this function

**Comment by da-anda** [comment]
IMO this change makes sense - a possible memory impact especially for embedded systems (PI, ARM) needs to be checked though. But if those can handle thousands of songs and their album covers in a list in same resolution, it shoulnd't be an issues for larger picture folders as well.

File: xbmc/settings/AdvancedSettings.h (hunk 1 above)
Comment (HitcherUK) [inline xbmc/settings/AdvancedSettings.h]: Thanks

**Comment by HitcherUK** [comment]
Well it looks like extracting of chapters has be broken on the RPi1 for some time anyway as it didn't work with this PR or without it. Even went back to OpenELEC-RPi.arm-6.0-Milhouse-20150515051606-#0514-gb1027e0.tar and it still doesn't work.
Creating bookmarks works fine with this PR though.

**Comment by da-anda** [comment]
did you make sure the setting is enabled on your PI?

**Comment by popcornmix** [comment]
Just to clarify the effects.

Texture (gpu) memory will be quadrupled for episode/chapter/bookmark thumbnails.
Not an issue for views that show a single thumbnail but if dozens are displayed in a wall view it may be a problem.
Each 1280x720 thumbnail will consume 4MB of texture memory.

The file sizes of the cached jpeg thumbnails will quadruple, which will have an effect on the loading time as well as the initial creation time.

By default on Pi the thumbnail creation is disabled, but can be enabled in video/file lists settings. As such I'm not so worried about this, but many Pi users do enable it, and I have been thinking about enabling it by default on Pi 2.

**Comment by HitcherUK** [comment]
Didn't even know that was an option on the Pi - will retest.

**Comment by HitcherUK** [comment]
The thumbs are actually 540*303 with this PR and 270*151 before.

Generating chapters takes ~12secs with and without this PR so there's no difference as far as I can see but would obviously like to hear other opinions.

Here's milhouse's test build for RPi1 - http://milhouse.openelec.tv/builds/testing/RPi/OpenELEC-RPi.arm-7.0-Milhouse-20160120094751-%230120i-gad66762.tar

**Comment by popcornmix** [comment]
> Generating chapters takes ~12secs with and without this PR so there's no difference as far as I can see but would obviously like to hear other opinions.

The time will be dominated by the software video decode (especially for high resolution video), compared to the jpeg encode (which is GPU accelerated on pi). This may be a little different on non-pi platforms, but I'd still expect the video decode to dominate and so the creation time won't change much with this PR.

What resolution in the thumb? I think OE sets a lower default than kodi's default (at least on Pi).

**Comment by HitcherUK** [comment]
> The thumbs are actually 540x303 with this PR and 270x151 before.

That's from a 16x9 1080p video file.

**Comment by HitcherUK** [comment]
Noticed that the thumbs in VideoOSDBookmarks weren't being loaded in the background so changed that as well.
Once the thumbs are generated there's no problem loading them on the RPi1.

**Comment by HitcherUK** [comment]
Also tested on Fire TV and no problems.

**Comment by HitcherUK** [comment]
Squashed.

**Comment by HitcherUK** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
ignore build fails. Good to merge?

**Comment by HitcherUK** [comment]
> ignore build fails. Good to merge?

I'd like an OK from other devs first - @da-anda @popcornmix ?

**Comment by popcornmix** [comment]
I'll abstain.
I have a slight concern over the increased memory and texture loading time.
But I won't object if others want this in.

**Comment by HitcherUK** [comment]
@popcornmix If it makes you feel better this will still be less memory than if you have a video playing in the background and you use Confluence's Thumbnail (10 thumbs) or Fanart (11 thumbs) views as the bookmark dialog only has 6 thumbs loaded.

**Comment by da-anda** [comment]
I ofc welcome the better picture quality, but it's not a must have if there are concerns. I'd say let's give it a try and if we notice issues, we still can revert or find another solution.

**Comment by MartijnKaijser** [comment]
I really can't see a downside cause posters+fanart+whatever in library view with 
>10 hogs way more memory

**Comment by HitcherUK** [comment]
@MartijnKaijser Give it a try?

**Comment by Razzeee** [comment]
maybe we can get this in the @MilhouseVH builds to get it tested?

**Comment by MilhouseVH** [comment]
Added it in build [#0120](http://forum.kodi.tv/showthread.php?tid=250817&pid=2221225#pid2221225)

**Comment by Razzeee** [comment]
Do you know if any problems came up?

**Comment by MilhouseVH** [comment]
None so far, although I doubt it's been tested much as - if I read this PR correctly - it doesn't affect video/music artwork (ie. posters, fanart etc.) but other more niche artwork types (pictures, bookmarks).

I've tested with some bookmark pictures on a RPi2 and haven't seen any problems - it wasn't super fast but bearable, pulling images from a 1080p movie (2.40:1 aspect ratio) using 720 <imageres> setting, generating 720x299 cached images (2.40:1 ratio)..

**Comment by HitcherUK** [comment]
@MilhouseVH I found generating images slow before this PR anyway. How does it load when you have generated images?

**Comment by MilhouseVH** [comment]
Sorry @HitcherUK, I hadn't realised you were waiting for a reply!

> How does it load when you have generated images?

Cached bookmark/chapter images are loading instantly.

Reason I'm back here... could you rebase this after https://github.com/xbmc/xbmc/commit/c27f9b000560c860714b362ebf2d35218ac8ac2b - thanks.

**Comment by MilhouseVH** [comment]
Something seems to have gone awry with the rebase... if you pull this patch (https://github.com/xbmc/xbmc/pull/8923.patch) it still has the old version (note the presence of `bool m_useDDSFanart;`).

**Comment by HitcherUK** [comment]
I'm not sure either as it's not in this branch -

https://github.com/HitcherUK/xbmc/blob/replace_thumb_size_with_image_size/xbmc/settings/AdvancedSettings.h

**Pull request closed without merging**